web/yo/app/scripts/services/mainutils.js

```diff
@@ -87,15 +87,42 @@ angular.module('oncokbApp')
             }
             var list = [];
             _.each(cancerTypes, function(cancerType) {
-                if (cancerType.subtype) {
-                    list.push(cancerType.subtype.name);
-                } else if (cancerType.mainType) {
-                    list.push(cancerType.mainType.name);
+                if (cancerType.mainType !== null && cancerType.mainType.name) {
+                    if (cancerType.subtype && cancerType.subtype.name) {
+                        list.push(cancerType.subtype.name);
+                    } else {
+                        list.push(cancerType.mainType.name);
+                    }
                 }
             });
             list = _.uniq(list).sort();
             return list.join(', ');
         }
+
+        function hasDuplicateCancerTypes(cancerTypes) {
+            var result = false;
+            if (!cancerTypes) {
+                result = false;
```

Comment:
change to "return false"
otherwise it will keep going to the _.some() function and throw error